packages/caip/src/adapters/thortrading/index.ts

```diff
@@ -0,0 +1,70 @@
+import { fromAssetId, toAssetId } from '../../assetId/assetId'
+import { btcAssetId, ethAssetId, ethChainId } from '../../constants'
+import { AssetId } from './../../assetId/assetId'
+import { btcChainId } from './../../constants'
+
+// https://regex101.com/r/d3Lgl9/1
+const thorIdRegex =
+  /(?<THORChainChain>[A-Z]+)\.(?<symbol>[A-Z]+)(-(?<contractAddress>[A-Z0-9]{42}))?/
+
+/*
+ * Note: this only works with thorchain assets we support
+ * see https://dev.thorchain.org/thorchain-dev/memos#asset-notation for reference
+ */
+export const poolAssetIdToAssetId = (id: string): AssetId | undefined => {
+  const matches = thorIdRegex.exec(id)?.groups
+  const THORChainChain = matches?.THORChainChain
+  const symbol = matches?.symbol
+
+  switch (THORChainChain) {
+    case 'ETH': {
+      if (id === 'ETH.ETH') return ethAssetId
+      if (THORChainChain === 'ETH' && symbol === 'ETH') return ethAssetId
+
+      const contractAddress = matches?.contractAddress
+      if (!contractAddress) return undefined
+
+      const chainId = ethChainId
+      const assetNamespace = 'erc20'
+      const assetReference = contractAddress.toLowerCase()
```

Comment:
while we can `toLowerCase`, `toAssetId` has validation and formatting included which we can chose to rely on instead (also includes contract address length check)

Reply:
wrap in `try/catch` and return undefined. Warning log?

Reply:
the `toLowerCase` is needed in this instance because of contract address check here (https://github.com/shapeshift/lib/blob/main/packages/caip/src/assetId/assetId.ts#L99)

Reply:
Feels like we should have this line: https://github.com/shapeshift/lib/blob/79a64fe24c74eb1ff0e778e4cb9795fed54fc11e/packages/caip/src/assetId/assetId.ts#L107 above that check then imo. Not directly related to this PR at this point though...